src/nn/visit_tensors.rs

```diff
@@ -0,0 +1,177 @@
+use crate::shapes::Dtype;
+
+use crate::prelude::*;
+use std::{
+    fmt::Debug,
+    string::{String, ToString},
+};
+
+pub struct ModuleGroup<'a, const N: usize, const M: usize, T> {
```

Comment:
What's the reasoning behind this struct? I think each module can just call visit separately on each tensor it holds, which would get rid of the need for this

Reply:
This is for "zipped" visiting, it holds a group of N immutable and M mutable references to a module, and has a convenience method for creating a new ModuleGroup that holds references to a field of each referenced Module.

Reply:
Ohhh I see, that's very clever!